src/Metadata/ValidatableClass.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Tuf\Metadata;
+
+use function DeepCopy\deep_copy;
+
+/**
+ * Defines a class that can used with Symfony validator constraints.
+ *
+ * Symfony validator constraints do not work with classes that have dynamic
+ * public properties.
+ */
+final class ValidatableClass extends \ArrayObject
```

Comment:
This class is really simple now. We could use ArrayObject directly except that we want to make sure that `offsetGet()` always returns clones so that the metadata is not altered after it is set.

Reply:
So the reason we needed this was because `\Tuf\KeyDB::createFromRootMetadata` was adding info to the key info. This didn't matter before but because this is now an object it is returned by reference. So the idea here is it that our metadata should not be change when a getter is used. 

but I think it is better to handle this on the getters for the metadata. So I am going to add `deep_copy()` there. This will allow remove this class completely and just `ArrayObject` everywhere we were using this.